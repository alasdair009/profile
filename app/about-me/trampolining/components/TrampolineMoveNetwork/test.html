<html lang="en">
  <head>
    <style>
      .videos {
        display: flex;
        height: 100vh;
        width: 100vw;
      }
      .videos div {
        align-items: center;
        display: flex;
        flex-direction: column;
        justify-content: center;
        width: 50%;
      }
      iframe {
        aspect-ratio: 16/9;
        width: 90%;
      }
    </style>
  </head>
  <body>
    <div class="videos">
      <div>
        <iframe
          id="standard"
          src=""
          data-src="https://www.youtube.com/embed/NpEaa2P7qZI?si=uJjK4C2TLWivF5BX"
          title="Standard Embed"
          allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share"
          referrerpolicy="strict-origin-when-cross-origin"
          allowfullscreen
        ></iframe>
        <button id="goStandard">Standard</button>
      </div>
      <div>
        <iframe
          id="privacy"
          src=""
          data-src="https://www.youtube-nocookie.com/embed/NpEaa2P7qZI?si=uJjK4C2TLWivF5BX"
          title="Enhanced privacy"
          allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share"
          referrerpolicy="strict-origin-when-cross-origin"
          allowfullscreen
        ></iframe>
        <button id="goPrivacy">Privacy</button>
      </div>
    </div>
    <script>
      const goStandard = document.getElementById("goStandard");
      const standardIFrame = document.getElementById("standard");
      const goPrivacy = document.getElementById("goPrivacy");
      const privacyIFrame = document.getElementById("privacy");
      goStandard.addEventListener(
        "click",
        function () {
          standardIFrame.setAttribute("src", standardIFrame.getAttribute("data-src"));
        },
        false
      );
      goPrivacy.addEventListener(
        "click",
        function () {
          privacyIFrame.setAttribute("src", privacyIFrame.getAttribute("data-src"));
        },
        false
      );
    </script>
  </body>
</html>
