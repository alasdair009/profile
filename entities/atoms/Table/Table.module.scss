@use "../../../app/styles/colors";
@use "../../../app/styles/dimensions";
@use "../../../app/styles/typography";
@use "../../../app/styles/helpers/rem";
@use "sass:color";
@use "sass:map";

.root {
  background: colors.$colorblackEvil;
  border-collapse: collapse;
  margin: rem.toRem(map.get(dimensions.$sizes, s16)) auto;
  max-width: rem.toRem(dimensions.$globalContentMaxWidth);
  vertical-align: middle;
  width: 100%;

  thead,
  tfoot {
    background: colors.$colorGreenGrass;
  }

  td {
    border: rem.toRem(1) solid colors.$colorGreyDark;
    padding: rem.toRem(map.get(dimensions.$sizes, s8));
    position: relative;
  }

  th {
    font-weight: map.get(typography.$fontWeights, bold);
  }

  @each $key, $width in dimensions.$breakpoints {
    &#{$key} {
      @media (max-width: #{$width}px) {
        table,
        thead,
        tbody,
        tfoot,
        th,
        td,
        tr {
          display: block;
        }

        thead,
        tfoot {
          background: none;
        }

        thead tr {
          position: absolute;
          top: -9999px;
          left: -9999px;
        }

        tr {
          border: rem.toRem(1) solid
            color.scale(colors.$colorGreyDark, $lightness: 5%);
        }

        td,
        th {
          border: none;
          border-bottom: rem.toRem(1) solid colors.$colorGreyDark;
          padding-left: 50%;
          position: relative;
          text-align: left;
        }

        td::before,
        th::before {
          content: attr(data-title) ":";
          font-weight: map.get(typography.$fontWeights, bold);
          left: rem.toRem(map.get(dimensions.$sizes, s8));
          position: absolute;
          top: 50%;
          transform: translateY(-50%);
          width: 45%;
          white-space: nowrap;
        }
      }
    }
  }
}
