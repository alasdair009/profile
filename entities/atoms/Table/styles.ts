"use client";
import styled from "styled-components";
import { Breakpoint, colors, device, globalContentMaxWidth, sizes } from "@/entities";
import { lighten, rem } from "polished";
import { breakpoints } from "@/entities/design-tokens/dimensions/dimensions";
import { typography } from "@storybook/theming";

export const Root = styled.table<{ $breakAt: Breakpoint }>`
  background: ${colors.blackEvil};
  border-collapse: collapse;
  margin: ${sizes.s16.rem} auto;
  max-width: ${rem(globalContentMaxWidth)};
  vertical-align: middle;
  width: 100%;

  thead,
  tfoot {
    background: ${colors.greenGrass};
  }

  td {
    border: ${rem(1)} solid ${colors.greyDark};
    padding: ${sizes.s8.rem};
    position: relative;
  }

  th {
    font-weight: ${typography.weight.bold};
  }

  @media (max-width: ${({ $breakAt }) => breakpoints[$breakAt]}px) {
    table,
    thead,
    tbody,
    tfoot,
    th,
    td,
    tr {
      display: block;
    }

    thead,
    tfoot {
      background: none;
    }

    thead tr {
      position: absolute;
      top: -9999px;
      left: -9999px;
    }

    tr {
      border: ${rem(1)} solid ${lighten(0.5, colors.greyDark)};
    }

    td,
    th {
      border: none;
      border-bottom: ${rem(1)} solid ${colors.greyDark};
      padding-left: 50%;
      position: relative;
      text-align: left;
    }

    td::before,
    th::before {
      content: attr(data-title) ":";
      font-weight: ${typography.weight.bold};
      left: ${sizes.s8.rem};
      position: absolute;
      top: 50%;
      transform: translateY(-50%);
      width: 45%;
      white-space: nowrap;
    }
  }
`;
