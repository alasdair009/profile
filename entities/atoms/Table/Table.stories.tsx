import { Table } from "./Table";
import { Meta, StoryObj } from "@storybook/react";
import { expect, within } from "@storybook/test";

const meta: Meta<typeof Table> = {
  component: Table,
  args: {
    children: (
      <>
        <thead>
          <tr>
            <th>col 1 heading</th>
            <th>col 2 heading</th>
            <th>col 3 heading</th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td data-title="col 1 heading">col 1 row 1</td>
            <td data-title="col 2 heading">col 2 row 1</td>
            <td data-title="col 3 heading">col 3 row 1</td>
          </tr>
          <tr>
            <td data-title="col 1 heading">col 1 row 2</td>
            <td data-title="col 2 heading">col 2 row 2</td>
            <td data-title="col 3 heading">col 3 row 2</td>
          </tr>
          <tr>
            <td data-title="col 1 heading">col 1 row 3</td>
            <td data-title="col 2 heading">col 2 row 3</td>
            <td data-title="col 3 heading">col 3 row 3</td>
          </tr>
        </tbody>
        <tfoot>
          <tr>
            <th data-title="col 1 foot label">col 1 foot</th>
            <th data-title="col 2 foot label">col 2 foot</th>
            <th data-title="col 3 foot label">col 3 foot</th>
          </tr>
        </tfoot>
      </>
    ),
  },
};
export default meta;

export const Default: StoryObj<typeof Table> = {
  play: async ({ args, canvasElement }) => {
    const canvas = within(canvasElement);

    await expect(canvas.getByTestId(Table.name)).toBeInTheDocument();
    await expect(canvas.getByText("col 1 row 1")).toBeInTheDocument();
  },
};

export const Responsive: StoryObj<typeof Table> = {
  args: {
    breakAt: "xlarge",
  },
  parameters: {
    docs: {
      story: {
        inline: false,
        height: 600,
      },
    },
  },
  play: async ({ args, canvasElement }) => {
    const canvas = within(canvasElement);

    await expect(canvas.getByTestId(Table.name)).toBeInTheDocument();
    await expect(canvas.getByText("col 1 row 1")).toBeInTheDocument();
  },
};
