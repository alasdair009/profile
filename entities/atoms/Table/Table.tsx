import { Root } from "./styles";
import { HTMLAttributes } from "react";
import { Breakpoint } from "@/entities";

type TableProps = {
  /**
   * Below this breakpoint the table will render in a vertical view
   */
  breakAt?: Breakpoint;
} & HTMLAttributes<HTMLTableElement>;

/**
 * Rotating icon used to indicate something is loading or waiting.
 */
export function Table({ breakAt = "base", children, ...rest }: TableProps) {
  return (
    <Root $breakAt={breakAt} {...rest}>
      {children}
    </Root>
  );
}
