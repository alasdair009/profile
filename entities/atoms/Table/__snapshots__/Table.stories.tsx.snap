// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`atoms/Table Default play-test 1`] = `
<table data-testid="Table"
       class="sc-hDcvty cQUvcm"
>
  <thead>
    <tr>
      <th>
        col 1 heading
      </th>
      <th>
        col 2 heading
      </th>
      <th>
        col 3 heading
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td data-title="col 1 heading">
        col 1 row 1
      </td>
      <td data-title="col 2 heading">
        col 2 row 1
      </td>
      <td data-title="col 3 heading">
        col 3 row 1
      </td>
    </tr>
    <tr>
      <td data-title="col 1 heading">
        col 1 row 2
      </td>
      <td data-title="col 2 heading">
        col 2 row 2
      </td>
      <td data-title="col 3 heading">
        col 3 row 2
      </td>
    </tr>
    <tr>
      <td data-title="col 1 heading">
        col 1 row 3
      </td>
      <td data-title="col 2 heading">
        col 2 row 3
      </td>
      <td data-title="col 3 heading">
        col 3 row 3
      </td>
    </tr>
  </tbody>
  <tfoot>
    <tr>
      <th data-title="col 1 foot label">
        col 1 foot
      </th>
      <th data-title="col 2 foot label">
        col 2 foot
      </th>
      <th data-title="col 3 foot label">
        col 3 foot
      </th>
    </tr>
  </tfoot>
</table>
`;

exports[`atoms/Table Responsive play-test 1`] = `
<table data-testid="Table"
       class="sc-hDcvty jdgvWd"
>
  <thead>
    <tr>
      <th>
        col 1 heading
      </th>
      <th>
        col 2 heading
      </th>
      <th>
        col 3 heading
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td data-title="col 1 heading">
        col 1 row 1
      </td>
      <td data-title="col 2 heading">
        col 2 row 1
      </td>
      <td data-title="col 3 heading">
        col 3 row 1
      </td>
    </tr>
    <tr>
      <td data-title="col 1 heading">
        col 1 row 2
      </td>
      <td data-title="col 2 heading">
        col 2 row 2
      </td>
      <td data-title="col 3 heading">
        col 3 row 2
      </td>
    </tr>
    <tr>
      <td data-title="col 1 heading">
        col 1 row 3
      </td>
      <td data-title="col 2 heading">
        col 2 row 3
      </td>
      <td data-title="col 3 heading">
        col 3 row 3
      </td>
    </tr>
  </tbody>
  <tfoot>
    <tr>
      <th data-title="col 1 foot label">
        col 1 foot
      </th>
      <th data-title="col 2 foot label">
        col 2 foot
      </th>
      <th data-title="col 3 foot label">
        col 3 foot
      </th>
    </tr>
  </tfoot>
</table>
`;
